### ISSUE TITLE
3.2.7: Subjective judgement is needed, and effort to test and remediate could be quite variable

### ISSUE THREAD

**Issue by benja11y** [w3c/wcag#1877]
The Understanding document says:

> Controls such as video players, web chats, and carousels include controls that are only visible on hover since they overlay the contents being displayed. These controls are not considered a process in terms of this SC but occasionally completing a process requires interacting with one of these controls.

We take this to mean that a page containing video thumbnails that can be hovered/focused to reveal a play button would still pass this SC—unless completing a process required it, in which case they would fail.

This means that subjective judgement is needed on whether persistently visible controls are required, based on what the user may be trying to accomplish, based on quite a subtle distinction. This could be prone to misinterpretation and/or variable results, as well as quite different outcomes in both design, as well as time required to remediate, for fairly similar UIs.

The SC has an exception for equivalent components, as seen below.
>The information needed to identify the user interface components is available through an equivalent component that is visible on the same page or on a different step in a multi-step process without requiring pointer hover or keyboard focus;
 
If a piece of text was added onto the page, stating that "video controls are revealed on mouse over and keyboard focus", should this pass the SC?

With the current wording requiring an "equivalent component", we do not think it currently passes, but suggest that the SC could be given a [labels or instructions](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html) exception.

This issue has been raised on behalf of [TPGi](https://www.tpgi.com), based on an internal consultation process. [An article containing all of our comments](https://www.tpgi.com/wcag-22-working-draft-feedback/) can be found on our blog, along with an introduction and acknowledgments.

**Comment by alastc** [comment]
> If a piece of text was added onto the page, stating that "video controls are revealed on mouse over and keyboard focus", should this pass the SC?

I hadn't considered that scenario before, but I think that would count as:
"information needed to identify that user interface components are available is visible"

I'll raise with the COGA folks who lead the SC development.

**Comment by bruce-usab** [comment]
Thank you @benja11y for the thoughtful feedback.

Where is the quoted bit <q>controls are not considered a process in terms of this SC</q> from exactly?  I am not finding that in Understanding [stable](https://www.w3.org/WAI/WCAG22/Understanding/visible-controls.html) or [editors](https://w3c.github.io/wcag/understanding/visible-controls.html) version.  That bit does seem awkward to me, but I think I need more context.  I am also not clear regarding what changes you are recommending?  Or are you just asking for clarification regarding the new proposed SC?

@alastc it certain has face validity that text instructions about mouseover or tabbing would count as <q>information needed</q> -- so thanks very much for circling back to the COGA folks.

**Comment by lseeman** [comment]
we think clear instruction on how to reach the control, that are on the page before or adjacent to the control is enough

**Comment by mbgower** [comment]
@benja11y a number of changes have been made to the Understanding document since your issue was opened 5 months ago. For example, the passage you cite has now been modified to read:
> Information needed to identify user interface components is usually the component itself; however, there are situations where controls are not visible. Complex components such as video players and carousels often include controls that are visible on hover. The presence of the complex component — for example, the carousel component or the video image — provides the “Information needed to identify the user interface components.” For a carousel, controls displayed on hover should also be displayed when the carousel is activated. For a video player, activating the video image should launch the video (sometimes in a new window), at which point more video player controls, including controls displayed previously on hover, should be visible. Some users may still struggle if video controls are not persistently visible, so there is benefit to providing a mechanism that allows users to have the controls persistently visible.

We believe this addresses some of your concerns with misinterpretation. Can you please verify?

Your second question asked if text instructions could be used to meet this requirement:
> If a piece of text was added onto the page, stating that "video controls are revealed on mouse over and keyboard focus", should this pass the SC?

You were concerned that text would not meet the "equivalent component" wording of the first bullet. However, as @alastc [points out](https://github.com/w3c/wcag/issues/1877#issuecomment-860272950), the preamble of the requirement itself allows text instructions to be sufficient:
> ...information needed to identify that user interface components are available is visible...

@lseeman pointed out in a [response](https://github.com/w3c/wcag/issues/1877#issuecomment-884920783) that the identifying information should be clear.

Obviously there is a degree of interpretation here, both in what is "needed to identify" and what is "clear". Authors concerned with differences of interpretation can seek to ensure hover or focus are never the triggers for making components visible, or that a mechanism exists to make the components persistently visible.

One thing I have noticed while replying to this issue is that the wording of bullet 3 should be about making the _components_ persistently visible, not the information (which needs to be visible already). I am creating a PR to address this.

**Comment by mbgower** [comment]
@alastc please review #2125 Although I think the logic for the change stems from my work to create an official response to this issue, the issue does not specifically call out this need. So if necessary, I can open a separate issue.

**Comment by alastc** [comment]
The response above was accepted, and there is a separate PR that has updated the SC text.
